lib/compress.js

```diff
@@ -338,7 +338,7 @@ merge(Compressor.prototype, {
                     if (ref.scope.uses_eval || ref.scope.uses_with) break;
 
                     // Constant single use vars can be replaced in any scope.
-                    if (var_decl.value.is_constant(compressor)) {
+                    if (!(var_decl.value instanceof AST_RegExp) && var_decl.value.is_constant(compressor)) {
```

Comment:
@kzc That use has nothing to do with it being "constant" (which it isn't); it can be reduced because it's the same AST. The same could be done with

```js
return x ? typeof a : typeof a;
```

```js
return x ? Math.pow(a, b) : Math.pow(a, b);
```


```js
return x ? function (a) { return 1 + a; } : function (a) { return 1 + a; };
```

etc.